api/types/registry/registry.go

```diff
@@ -102,3 +105,13 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// DistributionInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+type DistributionInspect struct {
+	// Digest is the content addressable digest for the image on the registry
+	Digest digest.Digest
```

Comment:
@stevvooe starting from a `distribution.Repository` object, it isn't clear how to get the `Descriptor` object cleanly in all cases. When the image has a tag (eg. `latest`), then the `Get(ctx, tag)` function for `TagService` can be used to get the `Descriptor`.

For images that only come in with a digest, the way seems to be calling `Blobs(ctx).Stat(ctx, digest)`, but that is causing an error. I wonder if it's worth adding extra information here that we don't intend to use right now.

cc @aaronlehmann